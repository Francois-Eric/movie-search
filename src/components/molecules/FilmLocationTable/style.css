.FilmLocationTable__container::-webkit-scrollbar {
  display: none;
}
.FilmLocationTable__container {
  max-height: 500px;
}
