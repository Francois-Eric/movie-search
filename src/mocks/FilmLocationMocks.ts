export const mockedFilmLocationRecords = [
  {
    id: "id_1",
    coord_x: 1.1,
    coord_y: 1.2,
    type_tournage: "type_tournage_1",
    nom_producteur: "nom_producteur_1",
    date_fin: "date_fin_1",
    nom_tournage: "nom_tournage_1",
    ardt_lieu: "ardt_lieu_1",
    id_lieu: "id_lieu_1",
    nom_realisateur: "nom_realisateur_1",
    adresse_lieu: "adresse_lieu_1",
    date_debut: "date_debut_1",
    annee_tournage: "annee_tournage_1",
  },
  {
    id: "id_2",
    coord_x: 2.1,
    coord_y: 2.2,
    type_tournage: "type_tournage_2",
    nom_producteur: "nom_producteur_2",
    date_fin: "date_fin_2",
    nom_tournage: "nom_tournage_2",
    ardt_lieu: "ardt_lieu_2",
    id_lieu: "id_lieu_2",
    nom_realisateur: "nom_realisateur_2",
    adresse_lieu: "adresse_lieu_2",
    date_debut: "date_debut_2",
    annee_tournage: "annee_tournage_2",
  },
];

export const mockedFilmLocationFilters = [
  {
    id: "filter_1",
    values: [
      { name: "filter_1_1", count: 2 },
      { name: "filter_1_2", count: 7 },
      { name: "filter_1_3", count: 5 },
    ],
  },
  {
    id: "filter_2",
    values: [
      { name: "filter_2_1", count: 8 },
      { name: "filter_2_2", count: 1 },
      { name: "filter_2_3", count: 6 },
      { name: "filter_2_4", count: 3 },
    ],
  },
];

export const mockedFilmLocationSelectedFilters = {
  filter_2: ["filter_2_2"],
};
