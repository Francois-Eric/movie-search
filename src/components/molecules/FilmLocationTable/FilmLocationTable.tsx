import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";
import { memo, RefObject } from "react";
import CONFIG from "../../../config.json";
import { FilmLocationNormalizedField } from "../../../types";
import { TableRowLoader } from "../../atom/TableRowLoader/TableRowLoader";
import "./style.css";

interface RowProps {
  record: FilmLocationNormalizedField;
  fieldList: {
    id: string;
    label: string;
    filterable?: boolean;
  }[];
}

interface TableProps {
  records: FilmLocationNormalizedField[];
  isLoading?: boolean;
  tableContainerRef?: RefObject<HTMLDivElement>;
}
const FilmLocationRow = ({ record, fieldList }: RowProps) => (
  <TableRow>
    {fieldList.map((field) => (
      <TableCell key={`${record.id}_${field.id}`}>
        {record[field.id as keyof FilmLocationNormalizedField]}
      </TableCell>
    ))}
  </TableRow>
);

export const MemoizedRow = memo(FilmLocationRow);

export const FilmLocationTable = ({
  records,
  isLoading,
  tableContainerRef,
}: TableProps) => {
  const fieldList = CONFIG.api.filmLocation.fields;

  return (
    <Paper sx={{ width: "100%", overflow: "hidden" }}>
      <TableContainer
        ref={tableContainerRef || null}
        className={"FilmLocationTable__container"}
      >
        <Table stickyHeader>
          <TableHead>
            <TableRow>
              {fieldList.map((field) => (
                <TableCell key={field.id}>{field.label}</TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {records.map((record) => (
              <MemoizedRow
                record={record}
                fieldList={fieldList}
                key={record.id}
              />
            ))}
          </TableBody>
        </Table>
        {isLoading && <TableRowLoader />}
      </TableContainer>
    </Paper>
  );
};
