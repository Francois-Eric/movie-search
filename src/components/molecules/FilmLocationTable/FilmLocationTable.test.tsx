import { render } from "@testing-library/react";
import { mockedFilmLocationRecords } from "../../../mocks/FilmLocationMocks";
import { FilmLocationTable } from "./FilmLocationTable";

describe("<FilmLocationTable/>", () => {
  test("should render correctly", () => {
    const { container } = render(
      <FilmLocationTable records={mockedFilmLocationRecords} />
    );
    expect(container).toMatchSnapshot();
  });
});
