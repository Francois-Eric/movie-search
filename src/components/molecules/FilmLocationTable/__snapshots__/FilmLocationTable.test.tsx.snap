// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmLocationTable/> should render correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fw5qro-MuiPaper-root"
  >
    <div
      class="MuiTableContainer-root FilmLocationTable__container css-rorn0c-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Année
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Type
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Nom
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Réalisateur
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Producteur
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Adresse
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
              scope="col"
            >
              Code Postal
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              annee_tournage_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              type_tournage_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_tournage_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_realisateur_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_producteur_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              adresse_lieu_1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              ardt_lieu_1
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              annee_tournage_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              type_tournage_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_tournage_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_realisateur_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              nom_producteur_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              adresse_lieu_2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              ardt_lieu_2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
